### ISSUE TITLE
pydrake runtime error: Mosek is not installed in your build. You'll need to use a different solver.

### ISSUE THREAD

**Issue by weiqiao** [RobotLocomotion/drake#7583]
I'm calling Mosek solver in python as in drake/drake/bindings/pydrake/test/testMosekSolver.py
The line 
```
result = solver.Solve(prog)
```
triggers the runtime error
```
RuntimeError: Mosek is not installed in your build. You'll need to use a different solver.
```

I checked that Mosek works in C++ interface.

**Comment by weiqiao** [comment]
cc @hongkai-dai

**Comment by weiqiao** [comment]
cc @jwnimmer-tri

**Comment by soonho-tri** [comment]
@weiqiao , what do you have when you run `bazel build //bindings/pydrake/solvers/... --config=mosek && ./bazel-bin/bindings/pydrake/solvers/mosek_solver_test`?

**Comment by weiqiao** [comment]
I got error 
```
ERROR: Skipping '//bindings/pydrake/solvers/...': no targets found beneath 'bindings/pydrake/solvers'

```

**Comment by soonho-tri** [comment]
Recently we removed `drake` directory and moved every sub-directory one-level up. Can you check-out the latest version and re-test it?

**Comment by jamiesnape** [comment]
Yes, we probably need to fix up the RPATH on the MOSEK library that we download (and again if we copied it to the install prefix). GUROBI is a little more awkward as it is not in the sandbox. We could fix up the RPATH, but it might be seen as impolite to modify something outside of the sandbox. There would be no issues on Mac, though.

**Comment by jamiesnape** [comment]
Actually ignore that. We would be modified the Drake shared objects, not Gurobi, so it would be fine.

**Comment by jamiesnape** [comment]
Opened #7592 which would remove the `Experimental` notation no the `WITH_GUROBI` and `WITH_MOSEK` options in CMake.

**Comment by EricCousineau-TRI** [comment]
Should be addressed by #7598 and #7873.
